src/GafferSceneUI/StandardLightVisualiser.cpp

```diff
@@ -855,3 +704,143 @@ IECoreGL::ConstRenderablePtr StandardLightVisualiser::cylinderRays( float radius
 
 	return group;
 }
+
+// Shading state management
+
+const char *StandardLightVisualiser::faceCameraVertexSource()
+{
+	return
+
+		"#version 120\n"
+		""
+		"#if __VERSION__ <= 120\n"
+		"#define in attribute\n"
+		"#define out varying\n"
+		"#endif\n"
+		""
+		"uniform int aimType;"
+		""
+		"uniform vec3 Cs = vec3( 1, 1, 1 );"
+		"uniform bool vertexCsActive = false;"
+		""
+		"in vec3 vertexP;"
+		"in vec3 vertexN;"
+		"in vec2 vertexuv;"
+		"in vec3 vertexCs;"
+		""
+		"out vec3 geometryI;"
+		"out vec3 geometryP;"
+		"out vec3 geometryN;"
+		"out vec2 geometryuv;"
+		"out vec3 geometryCs;"
+		""
+		"out vec3 fragmentI;"
+		"out vec3 fragmentP;"
+		"out vec3 fragmentN;"
+		"out vec2 fragmentuv;"
+		"out vec3 fragmentCs;"
+		""
+		"void main()"
+		"{"
+		""
+		""
+		"	vec3 aimedXAxis, aimedYAxis, aimedZAxis;"
+		"	if( aimType == 0 )"
+		"	{"
+		"		vec4 viewDirectionInObjectSpace = gl_ModelViewMatrixInverse * vec4( 0, 0, -1, 0 );"
+		"		vec3 viewCross = cross( viewDirectionInObjectSpace.xyz, vec3( 0, 0, -1 ) );"
+		"		aimedYAxis = length( viewCross ) > 0.0001 ? normalize( viewCross ) : vec3( 1, 0, 0 );"
+		"		aimedXAxis = normalize( cross( aimedYAxis, vec3( 0, 0, -1 ) ) );"
+		"		aimedZAxis = vec3( 0, 0, 1 );"
+		"	}"
+		"	else"
+		"	{"
+		"		aimedXAxis = normalize( gl_ModelViewMatrixInverse * vec4( 0, 0, -1, 0 ) ).xyz;"
+		"		aimedYAxis = normalize( gl_ModelViewMatrixInverse * vec4( 0, 1, 0, 0 ) ).xyz;"
+		"		aimedZAxis = normalize( gl_ModelViewMatrixInverse * vec4( 1, 0, 0, 0 ) ).xyz;"
+		"	}"
+		""
+		"	vec3 pAimed = vertexP.x * aimedXAxis + vertexP.y * aimedYAxis + vertexP.z * aimedZAxis;"
+		""
+		"	vec4 pCam = gl_ModelViewMatrix * vec4( pAimed, 1 );"
+		"	gl_Position = gl_ProjectionMatrix * pCam;"
+		"	geometryP = pCam.xyz;"
+		"	geometryN = normalize( gl_NormalMatrix * vertexN );"
+		"	if( gl_ProjectionMatrix[2][3] != 0.0 )"
+		"	{"
+		"		geometryI = normalize( -pCam.xyz );"
+		"	}"
+		"	else"
+		"	{"
+		"		geometryI = vec3( 0, 0, -1 );"
+		"	}"
+		""
+		"	geometryuv = vertexuv;"
+		"	geometryCs = mix( Cs, vertexCs, float( vertexCsActive ) );"
+		""
+		"	fragmentI = geometryI;"
+		"	fragmentP = geometryP;"
+		"	fragmentN = geometryN;"
+		"	fragmentuv = geometryuv;"
+		"	fragmentCs = geometryCs;"
+		"}"
+
+	;
+}
+
+const char *StandardLightVisualiser::texturedConstantFragSource()
+{
+	return
+		"#version 120\n"
+		""
+		"#if __VERSION__ <= 120\n"
+		"#define in varying\n"
+		"#endif\n"
+		""
+		"#include \"IECoreGL/ColorAlgo.h\"\n"
+		""
+		"in vec2 fragmentuv;"
+		""
+		"uniform vec3 lightMultiplier;"
+		"uniform vec3 defaultColor;"
+		"uniform float previewOpacity;"
```

Comment:
@danieldresser Are you using these in IELights? We don't use `lightMultiplier` or `previewOpacity` here, but didn't want to remove them if you were using them.